package BenchMark;
option optimize_for = LITE_RUNTIME;

//-10
message ReqReg{
	required string name  = 1;
	required string pwd  = 2;
	required int32 charId = 3;
}

//-1010
message ResReg{
	required int32 status = 1;
}

//-11
message ReqLogin{
	required int32 charId = 1;
	required string pwd  = 2;
}

//-1011
message ResLogin{
	required int32 status = 1;//-21: wrong pwd, -22: no such user;
}

//-12
message ReqMove{
	required int32 charId = 1;
	required int32 x  = 2;
	required int32 y  = 3;
	required int32 time  = 4;
}

//-1012
message ResMove{
	required int32 charId = 1;
	required int32 x  = 2;
	required int32 y  = 3;
	required int32 time  = 4;
}

//-21
message ReqUdpLogin{
	required int32 charId = 1;
}

//-1021
message ResUdpLogin{
	required int32 status = 1;//-21: wrong pwd, -22: no such user;
}

//-22
message ReqUdpMove{
	required int32 charId = 1;
	required int32 x  = 2;
	required int32 y  = 3;
	required int32 time  = 4;
}

//-1022
message ResUdpMove{
	required int32 charId = 1;
	required int32 x  = 2;
	required int32 y  = 3;
	required int32 time  = 4;
}

